### PR TITLE
C++: Don't override getParameterSizeIndex in Accept

### PR DESCRIPTION
**Pull request by MathiasVP**
The model for the [accept](https://www.freebsd.org/cgi/man.cgi?query=accept) function incorrectly specified that the 3rd argument was a size parameter when in reality it's a _pointer_ to a size parameter. This mismatch broke IR construction on calls to accept.

### CODE REVIEW

File: cpp/ql/src/semmle/code/cpp/models/implementations/Accept.qll [hunk 1]
```diff
@@ -46,8 +46,8 @@ private class Accept extends ArrayFunction, AliasFunction, TaintFunction, SideEf
     i = 1 and buffer = false
   }
 
-  override ParameterIndex getParameterSizeIndex(ParameterIndex i) { i = 1 and result = 2 }
-
+  // NOTE: The size parameter is a pointer to the size. So we can't implement `getParameterSizeIndex` for
+  // this model.
```
Comment (geoffw0) [inline cpp/ql/src/semmle/code/cpp/models/implementations/Accept.qll:50]: Then is `hasArrayWithVariableSize` wrong as well?

**Review by geoffw0 — COMMENTED** [review]
Otherwise LGTM.

File: cpp/ql/src/semmle/code/cpp/models/implementations/Accept.qll (hunk 1 above)
Comment (MathiasVP) [reply cpp/ql/src/semmle/code/cpp/models/implementations/Accept.qll:48]: Good point. Fixed in 585606a9331ea97141a62e771080fcfdd1e601bf.

**Review by geoffw0 — APPROVED** [review]
:+1:

**Pull request merged**